### PR TITLE
[Feature] 2.0.2-prepare bug fix of Pressure tests #7511

### PR DESCRIPTION
**Pull request by zwZjut**
## Purpose of the pull request

https://github.com/apache/dolphinscheduler/issues/7511

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by zwZjut** [comment]
https://github.com/apache/dolphinscheduler/issues/7511

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java [hunk 1]
```diff
@@ -146,14 +149,14 @@ public void run() {
...
-
-            this.workerRegistryClient.handleDeadServer(workerZkPaths, NodeType.WORKER, Constants.DELETE_OP);
         } catch (Exception e) {
-            logger.error(e.getMessage(), e);
+            logger.error("worker registry error", e);
             throw new RuntimeException(e);
         }
 
+        // solve dead lock
+        logger.info(org.apache.dolphinscheduler.spi.utils.PropertyUtils.dumpProperties());
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java:158]: I can't get the point that why this logger.info can solve dead lock?

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java [hunk 2]
```diff
@@ -150,37 +151,76 @@ public void closeRegistry() {
...
             registryClient.getLock(failoverPath);
 
+            if (!registryClient.exists(path)) {
+                logger.info("path: {} not exists", path);
+                // handle dead server
+                registryClient.handleDeadServer(Collections.singleton(path), nodeType, Constants.ADD_OP);
+            }
+
+            //failover server
+            if (failover) {
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java:185]: The failove logic will be skipped if the master can't get the zk lock.

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java [hunk 3]
```diff
@@ -330,13 +435,21 @@ private void failoverMaster(String masterHost) {
...
             List<TaskInstance> validTaskInstanceList = processService.findValidTaskListByProcessId(processInstance.getId());
             for (TaskInstance taskInstance : validTaskInstanceList) {
                 if (Constants.NULL.equals(taskInstance.getHost())) {
                     continue;
                 }
+                if (taskInstance.getState().typeIsFinished()) {
+                    continue;
+                }
                 logger.info("failover task instance id: {}, process instance id: {}", taskInstance.getId(), taskInstance.getProcessInstanceId());
                 failoverTaskInstance(processInstance, taskInstance);
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java:458]: need to check by `checkTaskInstanceNeedFailover` before `failoverTaskInstance`.

Comment (zwZjut) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java:458]: line 399 ?

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java (hunk 1 above)
Comment (zwZjut) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java:158]: PropertyUtils is loaded when used before, now loaded in @PostConstruct

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java (hunk 2 above)
Comment (zwZjut) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java:185]: pick from https://github.com/apache/dolphinscheduler/pull/7475

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java (hunk 3 above)
Comment (zwZjut) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryClient.java:458]: add checkTaskInstanceNeedFailover

**Review by lenboo — APPROVED** [review]
+1

**Pull request merged**